watershed_workflow/river_tree.py

```diff
@@ -151,6 +151,34 @@ def merge(self):
         for child in self.children:
             parent.addChild(child)
 
+    def merge_with_child(self):
+        """Merges this with its one and only child."""
+        assert (len(self.children) == 1)
+
+        # fix properties
```

Comment:
isn't there already a function that does something like this?

Reply:
yes, our existing `merge` function merges a small node with its parents. However, there are cases, when it is desirable to merge with the child. In the simplify, we are calling the appropriate merge functions based on the situation.

Reply:
Then, rather than duplicating the code, can't you just call `merge(node.children[0])`?

Reply:
This avoids having multiple places to fix if we need to merge more attributes, for instance.

Reply:
Yes, that's an ingenious suggestion.

Reply:
although, the merged element will have properties of the smaller element. I guess it is fine!

Reply:
Just did some tests, and there is an incentive to keep properties of the larger reach. Some manipulations and workflow are based on some property attributes. Hence, node.children[0].merge() is not the same as node.merge_with_child(). I am not sure what is an elegant way to do this apart from calling defining merge_with_child().

Reply:
I would rather have one method with an option then, e.g.

```
def merge(self, properties_from='parent'):

    if properties_from == 'parent':
        # do all the things that are different
    elif properties_from == 'child':
        # do all the things that are different
    # do all the things that will be the same (e.g. combining contrib area, etc)
```

Reply:
I tried two methods explained below, both worked fine and yielded the same results. There was not much difference in code size of the two methods, so I settled on the second one, in which 'self' is always the node to be merged (node.segment is shorter than merge tolerance).

 Method 1: commit cc5d206b225cbdfe4d48b620847dda569b79e211
 Here we call merge by `node.children[0].merge(properties_from = 'self')`, telling to it merge parent into `self` and retain the properties from `self`, here 'self' is child of the shorter node. It did not save much code because combining contributing areas, moving coordinates, etc. were different when merging parent into 'self' and 'self' into parent. 

 Method 2: commit 8a26162f675686b8976f318954d53d3ce6ac6b99
Here we call merge by ` node.merge(to = 'child')`. In this method `self` is always the shorter node; and flag tells whether to merge 'self' to parent (default) or to child.

Reply:
I still don't understand  why this is necessary. 

What properties do you get that aren't well-posed for the merged thing that you must keep either the parent or child's properties for?  To me it just looks like we incompletely treat the properties when merging.

Reply:
Ethan and Saubhagya discussed it offline, below is the summary:

- The method2 approach has been retained. By default, it uses the original merge functionality, with an additional option to merge with the child element. This deviates from the standard merge definition.
- In the upcoming WW refactor, standard geometry manipulation methods will not include these specific peculiarities; instead, they will be handled separately.
- It's intended that IDs and other properties of the larger element are retained. We will maintain a list for each processed reach, which will include the IDs of each smaller reach merged into it.